### PR TITLE
Proof of concept image comparision code

### PR DESCRIPTION
**Pull request by dstansby**
Currently SunPy only compares hashes of the test figures it generates, which means it's impossible to tell what actually changed in the image. This PR is a proof of concept to allow comparing images that fail the tests.

The current flow is:

1.  Generate new image
2. Compare hash of new image to expected hash
3. If hash fails:
    - Save the new image
    - Download the expected image from an external repository (currently here: https://github.com/dstansby/sunpy-test-images)

Note this means all the figure tests now require remote data (the downside to having the images in an external repo). There may be a better way to do this that I haven't worked out...

### CODE REVIEW

File: sunpy/tests/figure_tests_env_3.5.yml [hunk 1]
```diff
@@ -17,13 +17,10 @@ dependencies:
 - fontconfig=2.12.1=2
 - freetype=2.5.5=2
 - glib=2.50.2=1
-- gst-plugins-base=1.8.0=0
-- gstreamer=1.8.0=0
```
Comment (dstansby) [inline sunpy/tests/figure_tests_env_3.5.yml]: I had to get rid of these to make the environment install, but the images generated fine without them.

Comment (nabobalis) [reply sunpy/tests/figure_tests_env_3.5.yml]: It would be nice if we could unpin as many items as possible in the future.

**Comment by nabobalis** [comment]
I like this concept. 

As requiring the net, well all the CI assume that so I don't see the issue.

**Comment by dstansby** [comment]
Yes, I think it does disappear (maybe there's a way to save stuff from a travis build though?).

Also woops, I forgot python 2 was a thing which is why everything is broken.

**Comment by dstansby** [comment]
Yay, having a image repo works! See https://github.com/dstansby/sunpy-test-images/commit/cdc0091139331f6bba28165d49992902e4950e2a for the difference between the old images and the new ones added in #2449

**Pull request closed without merging**